models/hymod/include/Hymod.h

```diff
@@ -127,12 +128,17 @@ class hymod_kernel
         // calculate et
         double et = calc_et(soil_m, et_params);
 
+        double groundwater_excess;  //excess water from groundwater reservoir that can be positive or negative
+        double excess_water;        // excess water from reservoir that can be positive or negative
+
         // get the slow flow output for this time - ks
-        double slow_flow = groundwater.response(slow, dt);
+        double slow_flow = groundwater.response_storage_meters(slow, dt, groundwater_excess);
+        slow_flow += groundwater_excess / dt;
```

Comment:
I'm not sure I like this.  The groundwater reservoir response is slow flow, but the excess isn't nessicarly slow flow, it really should go back to whatever produced slow.  Also, what happens when excess < 0?  At some point, someone has to reconcile the supply/demand mismatch.

Reply:
When excess < 0, then it would be subtracted. Do you have any suggestions on what to do with excess?

Reply:
I think that if something happens where the storage goes negative, then it indicates a solution overshoot.   The standard solution is to check if the change in storage this time step is negative and greater than the current storage, then we would make the change in storage equal to the current storage, current storage would be come zero, and we probably want to throw a warning that it happened.

Reply:
Thanks Fred. I accidentally passed over this comment when you made it. We added boundary checks for min/max and warnings and adjustments to storage to zero if below min and max if exceeding max. We drain the reservoir to min storage in that case by sending the remaining storage to output. We send to excess water back to the flux when over max.